nuxeo-core/nuxeo-core-convert-api/src/main/java/org/nuxeo/ecm/core/convert/api/ConversionException.java

```diff
@@ -46,4 +55,56 @@ public ConversionException(Throwable cause) {
         super(cause);
     }
 
+    public ConversionException(String message, BlobHolder blobHolder) {
+        super(message);
+        addInfos(blobHolder);
+    }
+
+    public ConversionException(String message, Throwable cause, BlobHolder blobHolder) {
+        super(message, cause);
+        addInfos(blobHolder);
+
+    }
+
+    public ConversionException(Throwable cause, BlobHolder blobHolder) {
+        super(cause);
+        addInfos(blobHolder);
+    }
+
+    public ConversionException(String message, Throwable cause, Blob blob) {
+        super(message, cause);
+        addInfos(Collections.singleton(blob));
+    }
+
+    public ConversionException(String message, Blob blob) {
+        super(message);
+        addInfos(Collections.singleton(blob));
+    }
+
+    protected void addInfos(BlobHolder blobHolder) {
+        if (blobHolder instanceof DocumentBlobHolder) {
+            DocumentBlobHolder documentBlobHolder = (DocumentBlobHolder) blobHolder;
+            addInfo(String.format("Document Id = %s", documentBlobHolder.getDocument().getId()));
+        }
+
+        Collection<Blob> blobs = new HashSet<>();
+        if (blobHolder.getBlob() != null) {
+            blobs.add(blobHolder.getBlob());
+        }
+
+        if (CollectionUtils.isNotEmpty(blobHolder.getBlobs())) {
+            blobs.addAll(blobHolder.getBlobs());
+        }
+
+        addInfos(blobs);
+    }
+
+    protected void addInfos(Collection<Blob> blobs) {
```

Comment:
i changes the code to simplify the code and avoids the use of Collections#singleton. But the use of HashSet is mandatory because in the case of org.nuxeo.ecm.core.api.blobholder.DocumentBlobHolde the getBlobs method returns the Main Blob. And it will be resulting of the same Blob two times in my colleection  that why i change the type to HashSet